src/main/java/org/wso2/carbon/identity/oauth2/grant/kerberos/KerberosGrant.java

```diff
@@ -189,7 +206,20 @@ else if (KerberosGrantConstants.KERBEROS_IDP_SPNPASSWORD.equals(property.getName
                         kerberosUsersId.lastIndexOf('@') :
                         kerberosUsersId.length();
                 AuthenticatedUser kerberosUser = new AuthenticatedUser();
-                kerberosUser.setUserName(kerberosUsersId.substring(0, indexOfAt));
+                String username = kerberosUsersId.substring(0, indexOfAt);
+                kerberosUser.setUserName(username);
+
+                if (StringUtils.isNotEmpty(userstoreDomains)) {
+                    String[] userStores = userstoreDomains.split(",");
+
+                    for (String userStoreDomain : userStores) {
+                        if (isUserExistsInUserStore(kerberosUser.getUserName(), tenantDomain, userStoreDomain)) {
+                            kerberosUser.setUserStoreDomain(userStoreDomain);
+                            break;
+                        }
+                    }
```

Comment:
What happens when the user doesn't exist in any of the specified domain?
Shouldn't we fail the request?

Reply:
So we have 2 scenarios. One where the IDP configs doesn't have user store domains and the other one where the user does not exist in any of the configured user stores.

For the first case, we will treat it as a federated scenario and fail grant validation for the second case.

Reply:
Fixed